### DISCUSSION TITLE
Better animations for snap area revealing and window resizing

### DISCUSSION THREAD

**Discussion by bunabyt3** [rxhanson/Rectangle#1071] [Ideas]
I plan to add animations to the windows when they are being resized by Rectangle and when the snap area shows up when you drag a window to the edge of the screen. I believe it will make the application look better and feel "smoother". I will work on this contribution further and send a pull request if my idea is approved.

**Comment by rxhanson** [comment]
Thanks for looking to contribute. Rectangle moves and resizes windows using the Accessibility API, and to do so properly and accounting for different displays, it must first resize the window, then move the window into place, then resize the window once more. Each of these commands can be done sequentially but not simultaneously. Animating this behavior wouldn’t feel as smooth as you might expect, unless the action can be reduced down to only one of these commands, i.e. going from something like maximize to left half on the same display. Definitely not trying to deter you, but this is a limitation that is good to know up front. Unless you were able to find a way around that limitation, it’s not really a change I feel is worth merging in. With that said, feel free to fork it and see what you can come up with.
